types/highland/index.d.ts

```diff
@@ -734,6 +768,19 @@ declare namespace Highland {
 		 */
 		head(): Stream<R>;
 
+		/**
+		 * Creates a new Stream with the separator interspersed between the elements of the source.
+		 *
+		 * `intersperse` is effectively the inverse of [splitBy](#splitBy).
+		 *
+		 * @id intersperse
+		 * @section Transforms
+		 * @name Stream.intersperse(sep)
+		 * @param {R} sep - the value to intersperse between the source elements
+		 * @api public
+		 */
+		intersperse(separator: R): Stream<R>;
```

Comment:
I suppose strictly speaking there is no reason the intersperse element has to be of the same type as R. What would your thoughts be on this?

```
		intersperse<T>(separator: T): Stream<R | T>;
```

With these tests?

```
// $ExpectType Stream<Foo>
fooStream = fooStream.intersperse(foo);

// $ExpectType Stream<Foo|Bar>
fooStream.intersperse(bar)
```

Reply:
That makes sense. I honestly hadn't thought about it for cases besides strings, but that's easy to add and doesn't interfere with normal consumption.